src/include/common/logger.h

```diff
@@ -0,0 +1,65 @@
+#ifndef BISCUIT_COMMON_LOGGER_H
+#define BISCUIT_COMMON_LOGGER_H
+
+#include <memory>
+#include <string>
+
+#include "spdlog/sinks/basic_file_sink.h"
+#include "spdlog/spdlog.h"
+
+namespace biscuit::common {
+
+/**
+ * log file path.
+ */
+const char LOG_FILE[] = "logs/biscuitdb.log";
+
+/**
+ * LoggerConfig initializes spdlog to print debug messages and async logging.
+ */
+class LoggerConfig {
+  /* Flush to disk every 3 seconds */
+  const int ASYNC_FLUSH_TIME = 3;
+
+ public:
+  LoggerConfig();
+};
+
+/**
+ * Logger class for debuging.
+ * Note:
+ * This will not log debug messages when build type is Release.
+ * Other log messages are printed regardless.
+ * Use debug() where applicable for optimal performance in production.
+ */
+class Logger {
+  static const std::shared_ptr<spdlog::logger> LOG;
+  static const LoggerConfig CONFIG;
+
+ public:
+  /**
+   * Logs debug message.
+   */
+  static void Debug(const std::string &message);
+  /**
+   * Logs debug message.
+   */
+  static void Info(const std::string &message);
+  /**
+   * Logs debug message.
+   */
+  static void Warn(const std::string &message);
+  /**
+   * Logs debug message.
+   */
+  static void Error(const std::string &message);
+};
```

Comment:
Add 2 More Methods `GetLogger` & `RegisterLogger`

Reply:
What does Register
Logger do?

Reply:
To change the logger object. It's very likely to use different spdlog objects depending upon db setup configuration.

Reply:
I suggest we derive from this class for such cases. As we are using static objects.
Example:
```
class SpecialLogger: public Logger
```

As we will have multiple threads this needs to be thread safe. Registering logger from thread_i may cause unexpected behaviour in thread_j.

Reply:
Hi, good suggestion. But why redo the work when we are already using spdlog package. Also, users are never going to play with our logger package. The primary goal is to accommodate different logging needs passed during DB configuration.
For thread safety, we could use an RWmutex (instead of directly referencing LOG from Debug, Info... acquire a ReadLock and use the GetLogger).